### PR TITLE
Enable RELEASE=1 testing on CircleCi

### PR DESCRIPTION
**Pull request by wilzbach**
For some reason CircleCi doesn't immediately run the PR (maybe because there's no `appveyor.yml` on the target branch yet), so I had to setup it for my fork:

image

https://circleci.com/gh/wilzbach/dlang.org/84

Anyhow, we are all green now.

### CODE REVIEW

File: circleci.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            exit 1
+        fi
+    done
+}
+
+install_make()
+{
+    mkdir -p make
+    cd make
+    curl -L http://ftp.gnu.org/gnu/make/make-4.1.tar.gz  | tar -zxf - --strip-components=1 && ./configure && make && ./make -v
```
Comment (wilzbach) [inline circleci.sh]: The preinstalled make 3.8 gives a couple of weird errors.

File: circleci.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        local current_branch=$(git rev-parse --abbrev-ref HEAD)
+        git config user.name dummyuser
+        git config user.email dummyuser@dummyserver.com
+        git remote add upstream https://github.com/dlang/phobos.git
+        git fetch upstream
+        git checkout -f upstream/$base_branch
+        git merge -m "Automatic merge" $current_branch
+    fi
+
+    for proj in dmd druntime phobos ; do
```
Comment (wilzbach) [inline circleci.sh]: At a later stage we might want to test whether auto-cloning of all repos works, but for now this is the more bullet-proof approach and we can also checkout the same branch as the target branch if it exists.

File: circleci.sh [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+run_make()
+{
+    # Load environment for bootstrap compiler
+    source "$(CURL_USER_AGENT=\"$CURL_USER_AGENT\" bash ~/dlang/install.sh dmd-$HOST_DMD_VER --activate)"
+    export PATH="$(pwd)/make:$PATH"
+    make -v
+
+    # TODO: d-release.tag target is currently broken
+    #make -f posix.mak RELEASE=1 release -j5 html dmd-release druntime-release phobos-release d-release.tag
```
Comment (wilzbach) [inline circleci.sh]: So this already shows that setting up a CI is _very_ important. I will try to fix this in a follow-up PR once this is merged.

**Comment by CyberShadow** [comment]
>  ci/circleci — No test commands were found 

What's up?

**Comment by wilzbach** [comment]
> What's up?

Found the problem:

image

Apparently building PRs needs to be explicitly enabled.

**Pull request merged**